Customizing/global/plugins/Services/Repository/RepositoryObject/CareerGoal/classes/Observations/Observation.php

```diff
@@ -48,6 +48,22 @@ public function __construct($obj_id, $career_goal_id, $title, $description, $pos
 		$this->requirements = $requirements;
 	}
 
+	public function withObjectId($obj_id) {
+		assert('is_int($obj_id)');
+		$clone = clone $this;
+		$clone->obj_id = $obj_id;
+
+		return $clone;
+	}
+
+	public function withCareerGoalId($career_goal_id) {
+		assert('is_int($career_goal_id)');
+		$clone = clone $this;
+		$clone->career_goal_id = $career_goal_id;
+
+		return $clone;
+	}
```

Comment:
Ich finde diese beiden Methoden irritierend, da man imho vom Modell her die Beobachtungen nicht zu einem anderen Objekt umziehen kann. Ich glaube, die beiden Methoden gibt es nur aus technischen Gründen, da sie benötigt werden, um in der DB die Kopie eines Objekts durchzuführen. Ich denke aber, dass die Methoden dort nicht gebraucht werden, weil auch einfach die id des neuen Objekts bei Insert übergeben werden könnte.